flocker/control/testtools.py

```diff
@@ -341,7 +371,7 @@ def deployment_strategy(
         draw(
             lease_strategy(
                 dataset_id=st.just(dataset_id),
-                node_uuid=st.just(node_uuid)
+                node_id=st.just(node_uuid)
```

Comment:
Yep, it was broken and the node_strategy is also broken in the way it supplies `UUID(...)` versus `unicode(UUID(...))` to `Dataset.dataset_id`....I need to investigate that.
I just added the new `stateful_application` parameter to avoid triggering the bug in the existing tests.